New Releases (Dencun, ESM Refactor, Trie/SM Proof Functionality, WASM Crypto, Other) (#3261)
ethereumjs/ethereumjs-monorepo: packages/trie/CHANGELOG.md

There are also new static methods for the verification of proofs, namely `Trie.verifyProof()` to verify a proof for a single key as well as `Trie.verifyRangeProof()` for a range of keys.

Additionaly all proof related functionality is now better documented and there are README examples allowing for an easier entry to the topic.

Also note that along with this release round there is additional higher level proof functionality available shipped within the `@ethereumjs/statemanager` package.

### WASM Crypto Support

With this release round there is a new way to replace the native JS crypto primitives used within the EthereumJS ecosystem by custom/other implementations in a controlled fashion, see PR [#3192](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3192).

This can e.g. be used to replace time-consuming primitives like the commonly used `keccak256` hash function with a more performant WASM based implementation, see `@ethereumjs/common` [README](https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/common) for some detailed guidance on how to use.

### Self-Contained (and Working 🙂) README Examples

> All code examples in `EthereumJS` monorepo library README files are now self-contained and can be executed "out of the box" by simply copying them over and running "as is", see tracking issue [#3234](https://github.com/ethereumjs/ethereumjs-monorepo/issues/3234) for an overview. Additionally all examples can now be found in the respective library [examples](./examples/) folder (in fact the README examples are now auto-embedded from over there). As a nice side effect all examples are now run in CI on new PRs and so do not risk to get outdated or broken over time.

Reviewer: ```suggestion
All code examples in the `EthereumJS` monorepo library README files are now self-contained and can be executed "out of the box" by simply copying them over and running "as is", see tracking issue [#3234](https://github.com/ethereumjs/ethereumjs-monorepo/issues/3234) for an overview. Additionally all examples can now be found in their respective library [examples](./examples/) folder (in fact the README examples are now auto-embedded from over there). As a nice side effect, all examples are now run in CI on new PRs and so do not risk getting outdated or broken over time.
```

Author: Small nit: if you post suggestions for repeating text blocks (like the one above), maybe add a small note, also for others, so that a single one is not committed and then lead to issues on collective search-and-replace in the follow-up, remember that this has lead to some issues in the past (so apparently: the one changed text block would *then* be forgotten).

No big deal though, just to improve the process. 🙂
